### DISCUSSION TITLE
Suggestion to add info to installation guide - to avoid 404 Bad Request error on Proxmox installs

### DISCUSSION THREAD

**Discussion by itm1960** [homeassistant-apps/app-cloudflared#752] [Ideas]
For Proxmox users who get a "400 Bad Request" error when accessing your HA URL, it could be that you need to add the IP address of your Proxmox host to the list of trusted proxies - this fixed the problem for me. In my case I added 192.168.1.0/24 to the list.
Maybe this could be added to the installation guide/wiki?

**Comment by brenner-tobias** [comment]
Thanks a lot. I added a comment in the respecting section, which will be included after the PR (#825 ) is merged.

**Comment by wassupluke** [comment]
> For Proxmox users who get a "400 Bad Request" error when accessing your HA URL, it could be that you need to add the IP address of your Proxmox host to the list of trusted proxies - this fixed the problem for me. In my case I added 192.168.1.0/24 to the list.

Thank you for this, it helped me as well (plus a restart of homeassistantOS) :+1:

I'd add that the docs could be more clear that Proxmox users likely have to also add the CIDR IP for the host machine.

This add-on is a game-changer. Thank you!

**Reply by elcajon** [reply]
Could you explain to me what kind of setup you have? I also have HAOS running on a Proxmox VM and the default settings for the proxy work for me without any problems. I would like to understand where the differences are in our Proxmox systems that lead to you having to store other proxy IPs in HA core configuration.

**Reply by wassupluke** [reply]
Oh gosh, let's see if I can explain my setup.

I have a cluster of Proxmox nodes setup with the default networking (I don't have a separate network/VLAN for the cluster).

On one of the nodes I installed HAOS from the community post install [script](https://community-scripts.github.io/ProxmoxVE/scripts?id=haos-vm).

iirc, I installed it by simply letting it set the defaults for everything. If I didn't, then the only thing I would have changed is just setting my own password as the admin password and let the rest remain as defaults for the HAOS install.

Once logged in to the HAOS WebUI, I installed the built-in Cloudflare add-on that lives in the HAOS Store and connected my account. This didn't give me a tunnel like I wanted, so I googled around and found this repo. 

I then used the [add repo](https://my.home-assistant.io/redirect/supervisor_add_addon_repository/?repository_url=https%3A%2F%2Fgithub.com%2Fbrenner-tobias%2Fha-addons) button from the README here to install Cloudflared and setup my tunnel.

I've setup tunnels for other services running in my cluster without much issue (still haven't figured out how to get Minecraft to work as a subdomain that's still accessible from in the Minecraft launcher, but I think that's a problem to be solved by Warp or having end users install Cloudflared on their machine to authenticate with the tunnel, but that's another story). Anyway, I've done tunnels before and got them working easily. Not so much with running a tunnel into HAOS. iirc I had to customize the configuration file in HAOS for the tunnel to include not only the docker network CIDR IP, but also the CIDR IP of the host. Then everything worked as expected.

As I'm typing this I'm wondering if it's got something to do with the virtual bridge (the default `vmbr0` in Proxmox)?

I'm not at my computer right now to check other specifics but if there are other data points or configuration files of mine you'd like to see for troubleshooting, just let me know. 

All my school education is medical, I'm just messing with this stuff to try and teach myself some things about networking. Thanks for bearing with me as I learn. :)

**Reply by elcajon** [reply]
Thank you for the detailed feedback 😊

> I have a cluster of Proxmox nodes setup with the default networking

Do I understand correctly that you operate several Proxmox servers (=nodes) and run the VM with HAOS on one of them? If so, then that could possibly be the problem, as I don't know how load balancing works for VMs if the VM is operated via several nodes.

> Once logged in to the HAOS WebUI, I installed the built-in Cloudflare add-on that lives in the HAOS Store and connected my account. This didn't give me a tunnel like I wanted, so I googled around and found this repo.

I think you mean [this integration](https://www.home-assistant.io/integrations/cloudflare/)? It is only there to update the DNS entries with your homes public IP, but has nothing to do with the tunnel in this sense. So this add-on is the right thing for you. 

> I've setup tunnels for other services running in my cluster without much issue

Did you set them up using our add-ons `additional_hosts` parameter or do you run more than one cloudflared tunnel?

> (still haven't figured out how to get Minecraft to work as a subdomain that's still accessible from in the Minecraft launcher, but I think that's a problem to be solved by Warp or having end users install Cloudflared on their machine to authenticate with the tunnel, but that's another story)

Yea that's not that easy using Cloudflared tunnels, WARP would be the right thing to do if you want to use Cloudflare to get this solved --> https://developers.cloudflare.com/cloudflare-one/applications/non-http/
For use cases like this I would use Tailscale (there is an official HA add-on to get Tailscale up and running, pretty nice VPN in my opinion), but that's just my opinion

> As I'm typing this I'm wondering if it's got something to do with the virtual bridge (the default vmbr0 in Proxmox)

Have you played around with the default settings or are they still the default?

> All my school education is medical, I'm just messing with this stuff to try and teach myself some things about networking. 

Networks, especially when it comes to Docker and HA, are always quite special. If Proxmox is also running as a cluster with several nodes that communicate with each other, this naturally makes things even more complicated. 

I'm only asking because it should actually be easy in our case. HAOS provides a Docker network. Within the network, HA (which is actually the "HA Core" Docker container) communicates with the HA Supervisor (also a Docker container). In addition, all add-ons that you install are also started as Docker containers and all run on the same host (= HAOS = Proxmox VM) and share the CIDR 172.30.33.0/24 within Docker. So your instance of our cloudflared tunnel add-on should get an IP within this range which is the IP to communicate docker internal with your HA Core instance.

<img width="1008" alt="Bildschirmfoto 2025-05-11 um 23 05 08" src="https://github.com/user-attachments/assets/a25234e2-f186-4468-a71e-96d38881e79d" />

> Thanks for bearing with me as I learn. 😊

No problem at all, that's how I feel too, the nice thing about HA is that you learn something new every day. 😊

**Reply by elcajon** [reply]
@itm1960 feel free to share your config / experience as well. Maybe we can get this more resilient for all our users 😊
